provider/cloudflare.go

```diff
@@ -150,6 +155,26 @@ func (p *CloudFlareProvider) Zones(ctx context.Context) ([]cloudflare.Zone, erro
 	result := []cloudflare.Zone{}
 	p.PaginationOptions.Page = 1
 
+	// if there is a zoneIDfilter configured
+	// && if the filter isnt just a blank string (used in tests)
+	if len(p.zoneIDFilter.zoneIDs) > 0 && p.zoneIDFilter.zoneIDs[0] != "" {
```

Comment:
Maybe you could add a new method **ZoneDetailsByDomainFilter** which includes your code.

Reply:
or **ListZonesByIDFilter**

Reply:
I tried to split this off (and create functions that each returned iterators) but my minimal knowledge of Go wasn't enough to get things working.

Splitting off just the contents of this branch didn't seem to improve readability.